Fixes an issue with non-apple unix builds

MattSturgeon opened the pull request:
I can't test if everything still works on mac, windows and mingw but this fixes a regression where `unix.cpp` is no longer included in other-unix systems.

pafmaf commented on src/CMakeLists.txt:
```diff
@@ -2,8 +2,10 @@
     find_library(coreFoundation CoreFoundation)
     set(PLATFORM_TARGET_LIBS ${coreFoundation})
-elseif(MINGW)
+elseif(MINGW) # Should this be more general?
```
I think `imm32.lib` is only needed for `ImmGetContext` or some other Windows related thing, i.e. if you build with MinGW while using `win32.cpp` source, but I don't remember exactly.

pafmaf commented on src/openboardview/CMakeLists.txt:
```diff
@@ -41,18 +43,24 @@ set(SOURCES
+			osx.mm
+			unix.cpp
+		)
+	elseif (MINGW)
```
I think this one is unreachable since `UNIX` is false for `MINGW`.

MattSturgeon committed:
Fixes linux build system

Some changes broke cmake on linux, This fixes them.

Also tried to tidy up some system checking conditionals.

I cannot test if this breakes OSX, windows or MinGW though.

Signed-off-by: Matt Sturgeon <matt@sturgeon.me.uk>

MattSturgeon commented on src/CMakeLists.txt:
```diff
@@ -1,13 +1,15 @@
     set(PLATFORM_TARGET_LIBS ${coreFoundation})
-elseif(MINGW)
+elseif(MINGW) # Should this check for WIN32?
     set(PLATFORM_TARGET_LIBS libopengl32.a imm32.lib)
```
I didn't think the windows build used opengl

MattSturgeon commented on src/openboardview/CMakeLists.txt:
```diff
@@ -1,17 +1,19 @@
-find_package(SDL2 REQUIRED)
+find_package(SDL2 REQUIRED) # is this really required by windows?
```
If the windows build doesn't use SDL I can wrap this in a UNIX condition

MattSturgeon commented on src/openboardview/CMakeLists.txt:
```diff
@@ -41,19 +43,23 @@ set(SOURCES
-endif()
-
-if (MINGW)
+elseif (MINGW) # Should this just be WIN32?
```
I'm pretty sure it's better to just check for WIN32 here

MattSturgeon commented:
@pafmaf I've updated the commit, so should work for mingw and unix now.

I'm pretty sure most/all of the mingw conditions should be checking for win32 - this should help enable other windows generators like building for visualstudio.

FYI, I've also been doing some work and playing around [over here](/MattSturgeon/OpenBoardView/commits/cmake-changes)

pafmaf commented:
@MattSturgeon Thanks for that! I briefly checked and everything seems fine.

With MinGW the build also succeeds using `unix.cpp`, but then you need GTK3+ as lib (probably distributed with the .zip) and other stuff on Windows! So there's basically a few build flags missing in CMake (for Windows), ie. whether you want to use GTK+/native Win or OpenGL/DirectX etc.

Now, to keep the binary small and the build process simple on most platforms you'd avoid dragging all those libs (gtk+, opengl, on one hand) and maintaining many various different scenarios (VS, CLion, MacOS, Linux). So it all kind of depends on what the project is going for in the long run: Either MinGW only builds for Windows using SDL or additional support for VS and directX etc.

(Second option's not too cool, since we'd need to keep the ImGui stuff up to date and so forth.)
